### DISCUSSION TITLE
Recommendation: the Web UI uses storage api instead of cookies to save settings

### DISCUSSION THREAD

**Discussion by hanguokai** [transmission/transmission#4133] [Ideas]
Currently, the web ui uses cookies to save settings like 'sort-method'. Cookies have some limitations(defects), I suggest using other web storage api to save it, like `IndexedDB` or `localStorage`.

**Comment by g-regor** [comment]
I would prefer some server side storage.

Chrome already nuked my `localStorage` multiple times, because I ran out of disk space.

It would also be possible to share the same setting across multiple devices.

**Comment by killemov** [comment]
Upvote #6678.

**Comment by killemov** [comment]
If you are running several services behind a reverse proxy the browser will probably combine them. Usually the total size of cookies combined for a single site is round 4KB.  That is why the upcoming release of [Shift](https://github.com/killemov/Shift) will have the option of using localStorage instead of a cookie to store changes in the settings made in the browser. It was very easy to implement and that only took me a couple of minutes.
